### PR TITLE
add chirality to atom and bond features

### PR DESCRIPTION
**Pull request by xiongzhp**
three one-hot encoded chirality features, indicating CW(clockwise), CCW(Counter Clock Wise) and unspecified.

### CODE REVIEW

File: deepchem/feat/graph_features.py [hunk 1]
```diff
@@ -166,6 +166,9 @@ def atom_features(atom, bool_id_feat=False, explicit_H=False):
                              [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10]) + \
               one_of_k_encoding_unk(atom.GetImplicitValence(), [0, 1, 2, 3, 4, 5, 6]) + \
               [atom.GetFormalCharge(), atom.GetNumRadicalElectrons()] + \
+              one_of_k_encoding_unk(atom.GetChiralTag(),
```
Comment (lilleswing) [inline deepchem/feat/graph_features.py]: can we put this behind a feature flag at least to start?  
```python
use_chirality=False
```

Comment (lilleswing) [reply deepchem/feat/graph_features.py]: We shouldn't change the model default behaviour until we have reasonable proof that it does better on average for datasets we care about.

Comment (xiongzhp) [reply deepchem/feat/graph_features.py]: Yes, I can try that! Sorry for failing the test, I actually tested on my repo using Theano, it works fine. I didn't expect so tricky deepchem is!

Comment (xiongzhp) [reply deepchem/feat/graph_features.py]: Let me fix it

**Comment by proteneer** [comment]
It’s not clear to me if Chiral Tags are right way as a feature to go since they’re order dependent. (Intact I’m not even sure they work correctly with the reordering code). I’d be in favor of Cahn Ingle Prelog codes instead. You’d also need chirality for bonds.

**Comment by xiongzhp** [comment]
@proteneer you are right, but it seems that Chiral Tags are the only way to denote steric types, I just can't find more info about how it works. I think Chiral Tags should be unique for a specific atom in a specific molecule. As to chirality for bonds,  I don't think it is necessary because bond features are designed to characterize how atoms connect to each other. Atom chirality is enough to determine in which order its neighbor connect, so we don't need to encode that feature twice.

**Comment by proteneer** [comment]
>> It seems that Chiral Tags are the only way to denote steric types in rdkit, I just can't find more info about how it works. 

RDKit has CIP descriptors you can extract without too much difficulty.

>>> As to chirality for bonds, I don't think it is necessary because bond features are designed to characterize how atoms connect to each other. Atom chirality is enough to determine in which order its neighbor connect, so we don't need to encode that feature twice.

Bond and Atom Chirality are not duplicates of each other. In fact, a chiral double bond typically connects atoms that have no AtomChirality, and a chiral atom is typically connected to single achiral bonds. They're both needed.

**Comment by xiongzhp** [comment]
Many Thanks for the information and the kind explanation! I will try R/S and E/Z labeling for atoms and bonds respectively. I still need some time to figure it out.

**Comment by rbharath** [comment]
Tentatively, this PR looks good. Will leave open for a couple days in case there are more comments though.

**Comment by lilleswing** [comment]
@xiongzhp are you ready for this to be merged in?

**Comment by xiongzhp** [comment]
@lilleswing Yes

**Comment by rbharath** [comment]
LGTM

**Comment by rbharath** [comment]
@xiongzhp Congrats on a merged PR!

**Pull request merged**